src/cljs/doit/events.cljs

```diff
@@ -63,6 +81,44 @@
                  :on-success [::get-todos-success]
                  :on-failure [::request-failed]}}))
 
+(rf/reg-event-fx
+ ::update-todo
+ (fn [cofx [_ todo]]
+   (let [id (:id todo)
+         url (str todo-url (:id todo) "/")]
+     (prn url)
+     {:http-xhrio {:method          :put
+                   :uri             url
+                   :params          todo
+                   :timeout         8000
+                   :format          (ajax/json-request-format)
+                   :response-format (ajax/json-response-format {:keywords? true})
+                   :on-success      [::update-todo-success]
+                   :on-failure      [::request-failed]}})))
```

Comment:
why not `(defn update-todo)` here like the other events?

Reply:
I have created seperate `defn` for event handlers whom I intend to test. I didn't see much value in testing event handlers which purely makes an API call.